ivy_tests/test_ivy/test_frontends/test_jax/test_jax_random.py

```diff
@@ -1,203 +1,104 @@
 # global
-from hypothesis import strategies as st
-import ivy
+from hypothesis import strategies as st, given
+import jax
 
 # local
 import ivy_tests.test_ivy.helpers as helpers
-from ivy_tests.test_ivy.helpers import handle_frontend_test
+import ivy.functional.frontends.jax as jax_frontend
+
+"""
+Tests for jax.random cannot be made normally since a `uint32` PRNG key must
+be passed for jax, meaning torch and paddle would always fail.
+The format used below should be followed for jax.random tests
+"""
 
 
 @st.composite
 def _get_minval_maxval(draw):
     interval = draw(st.integers(min_value=1, max_value=50))
     minval = draw(st.floats(min_value=-100, max_value=100))
-    maxval = draw(st.floats(min_value=-100 + interval, max_value=100 + interval))
+    maxval = minval + interval
     return minval, maxval
 
 
-@handle_frontend_test(
-    fn_tree="jax.random.uniform",
-    dtype_key=helpers.dtype_and_values(
-        available_dtypes=["uint32"],
-        min_value=0,
-        max_value=2000,
-        min_num_dims=1,
-        max_num_dims=1,
-        min_dim_size=2,
-        max_dim_size=2,
-    ),
+@given(
+    key=st.integers(min_value=int(-1e15), max_value=int(1e15)),
     shape=helpers.get_shape(),
-    dtype=helpers.get_dtypes("float", full=False),
-    dtype_minval_maxval=_get_minval_maxval(),
+    dtype=helpers.get_dtypes("float", full=False, prune_function=False),
+    minval_maxval=_get_minval_maxval(),
 )
 def test_jax_uniform(
-    *,
-    dtype_key,
+    key,
     shape,
     dtype,
-    dtype_minval_maxval,
-    on_device,
-    fn_tree,
-    frontend,
-    test_flags,
+    minval_maxval,
 ):
-    input_dtype, key = dtype_key
-    minval, maxval = dtype_minval_maxval
-
-    def call():
-        return helpers.test_frontend_function(
-            input_dtypes=input_dtype,
-            frontend=frontend,
-            test_flags=test_flags,
-            fn_tree=fn_tree,
-            on_device=on_device,
-            test_values=False,
-            key=key[0],
-            shape=shape,
-            dtype=dtype[0],
-            minval=minval,
-            maxval=maxval,
-        )
-
-    ret = call()
-
-    if not ivy.exists(ret):
-        return
-
-    ret_np, ret_from_np = ret
-    ret_np = helpers.flatten_and_to_np(ret=ret_np)
-    ret_from_np = helpers.flatten_and_to_np(ret=ret_from_np)
-    for (u, v) in zip(ret_np, ret_from_np):
-        assert u.dtype == v.dtype
-        assert u.shape == v.shape
-
-
-@handle_frontend_test(
-    fn_tree="jax.random.normal",
-    dtype_key=helpers.dtype_and_values(
-        available_dtypes=["uint32"],
-        min_value=0,
-        max_value=2000,
-        min_num_dims=1,
-        max_num_dims=1,
-        min_dim_size=2,
-        max_dim_size=2,
-    ),
+    minval, maxval = minval_maxval
+
+    jax_frontend.config.update("jax_enable_x64", True)
```

Comment:
Wouldn't that still be enabled after the test finishes? maybe we should refactor the x64 enable code into a context manager and use it in JAX random tests so that after the test finishes it will clean up and reset the flag to False.